### ISSUE TITLE
Monster Hunter Frontier Z Client Crash

### ISSUE THREAD

**Issue by klark231** [ptitSeb/box86#720]
Box86 with Dynarec v0.2.7 09b234e7 built on Nov 14 2022 01:51:47
Box86 with Dynarec v0.2.7 09b234e7 built on Nov 14 2022 01:51:47
Box86 with Dynarec v0.2.7 09b234e7 built on Nov 14 2022 01:51:47
0088:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
008c:err:wineusb:usb_main_loop Failed to initialize libusb: Other error
0088:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0088:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0088:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
Box86 with Dynarec v0.2.7 09b234e7 built on Nov 14 2022 01:51:47
Box86 with Dynarec v0.2.7 09b234e7 built on Nov 14 2022 01:51:47
00dc:fixme:nstc:NSTC2_fnSetControlStyle2 mask & style (0x00000004) contains unsupported style(s): 0x00000004
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069F51C)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069F51C)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069F51C)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069F51C)
0074:fixme:imm:ImeSetActiveContext (00010026, 0): stub
0074:fixme:imm:ImmReleaseContext (00010020, 00010026): stub
00dc:fixme:imm:ImeSetActiveContext (0001004E, 1): stub
00dc:fixme:imm:ImmReleaseContext (00010056, 0001004E): stub
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (0016BB60)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (00173550)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (001741E8)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (00172938)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (00177AB0)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a003, 0, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1026, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a003, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnSendControlMsg stub, 0015FCC8 (2, 1025, a004, 1, 0069ECDC)
00dc:fixme:shell:IShellBrowser_fnOnViewWindowActive stub, 0015FCC8 (001726A0)
00dc:err:shell:HCR_GetFolderAttributes should be called for simple PIDL's only!
00dc:fixme:exec:SHELL_execute flags ignored: 0x0000000c
0148:fixme:heap:RtlSetHeapInformation handle 00000000, info_class 1, info 00000000, size 0 stub!
0148:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {12e1ddac-7ebb-434f-bc58-54c27d745f8f}
0148:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0148:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x30
0148:fixme:wbemprox:enum_class_object_Next timeout not supported
0148:fixme:d3d:wined3d_guess_card_vendor Received unrecognized GL_VENDOR "Collabora Ltd". Returning HW_VENDOR_NVIDIA.
0148:fixme:d3d:wined3d_guess_card_vendor Received unrecognized GL_VENDOR "Collabora Ltd". Returning HW_VENDOR_NVIDIA.
0148:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
TU: error: ../src/freedreno/vulkan/tu_image.c:273: FINISHME: UBWC with 3D textures

ALSA lib confmisc.c:855:(parse_card) cannot find card '0'
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory
ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default
ALSA lib confmisc.c:855:(parse_card) cannot find card '0'
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_card_inum returned error: No such file or directory
ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name
ALSA lib conf.c:5178:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5701:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2664:(snd_pcm_open_noupdate) Unknown PCM default
ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: No such file or directory
0148:err:winediag:MIDIMAP_drvOpen No software synthesizer midi port found, Midi sound output probably won't work.
0148:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x30
0148:fixme:d3d:wined3d_guess_card_vendor Received unrecognized GL_VENDOR "Collabora Ltd". Returning HW_VENDOR_NVIDIA.
0148:fixme:d3d:wined3d_guess_card_vendor Received unrecognized GL_VENDOR "Collabora Ltd". Returning HW_VENDOR_NVIDIA.
0148:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0148:fixme:ieframe:PersistStreamInit_InitNew (005066D8)
0148:fixme:ieframe:ViewObject_SetAdvise aspects and/or flags not supported yet
0148:fixme:ieframe:ViewObject_Draw (005066D8)->(1 -1 00000000 00000000 00000000 05010062 02747DE4 02747DE4 00000000 00000000)
0148:fixme:ieframe:WebBrowser_put_RegisterAsDropTarget (005066D8)->(0)
0148:fixme:urlmon:InternetBindInfo_GetBindString not supported string type 20
0148:fixme:ieframe:navigate_url Unsupported args (Flags 0021E5D4 {VT_I4: 2}; TargetFrameName 0021E594 {VT_EMPTY})
0148:fixme:urlmon:InternetBindInfo_GetBindString not supported string type 20
0148:fixme:ieframe:handle_navigation_error Navigate to error page
0148:fixme:imm:ImeSetActiveContext (000100CA, 1): stub
0148:fixme:imm:ImmReleaseContext (000300C8, 000100CA): stub
017c:fixme:iphlpapi:NotifyIpInterfaceChange (family 0, callback 077DACA0, context 011BB798, init_notify 0, handle 0CA7FDC0): stub
0148:fixme:imm:ImmGetDescriptionW (04090409, 00000000, 0): semi stub
0148:fixme:imm:ImmGetDescriptionW (04090409, 0F545188, 9): semi stub
0148:fixme:msctf:InputProcessorProfileMgr_GetActiveProfile (066756E0)->({34745c63-b2f0-4784-8b67-5e12c8701a31} 0021D704)
0148:fixme:file:NtLockFile I/O completion on lock not implemented yet
0148:fixme:ieframe:BrowserService_GetTravelLog 00526470 0021DD90
0148:fixme:ieframe:ClientSite_GetContainer (005067B0)->(0021DE8C)
0148:fixme:mshtml:nsCacheInfoChannel_GetCacheKey (00527920)->(0021D218)
0148:fixme:mshtml:nsCacheInfoChannel_GetCacheTokenExpirationTime (00527920)->(0021D228)
wine: Unhandled page fault on read access to 2E2E749C at address 0A091671 (thread 0148), starting debugger...
0148:err:seh:start_debugger Couldn't start debugger L"winedbg --auto 324 916" (1359)
Read the Wine Developers Guide on how to set up winedbg or another debugger

The client is based on Internet Explorer. it works on exagear. hope you can fix it on box86

**Comment by Pipetto-crypto** [comment]
"Unhandled page fault on read access to 2E2E749C at address 0A091671" I had a similiar error on a game, switching wine version helped. Make sure that the wine version you are using is also compatible with the game you are trying to run.

**Comment by klark231** [comment]
> "Unhandled page fault on read access to 2E2E749C at address 0A091671" I had a similiar error on a game, switching wine version helped. Make sure that the wine version you are using is also compatible with the game you are trying to run.
> 
> EDIT: To add to this, I just tested in exagear and Wine Internet explorer seems to crash as well there. Are you sure that it isn't running normal Internet Explorer? You can install normal Internet Explorer with winetricks

unfortunately I need internet explorer 8 or above for MHFZ client to run and for some reason winetricks doesnt work on arm64 android.

**Comment by Pipetto-crypto** [comment]
> I can confirm you a managed you to install ie8 on an arm64 android chroot

**Comment by klark231** [comment]
> I can confirm you a managed to install ie8 with winetricks on an arm64 android chroot and it does work ![Screenshot_2022-11-27_160205](https://user-images.githubusercontent.com/72308298/204142158-7c5e5ee2-22ad-43cf-97b4-f5d9e62dac26.jpg)

then I guess youre guide is incomplete. i tried your guide and winetricks doesnt work

**Comment by Pipetto-crypto** [comment]
I used this guide to install winetricks:

https://github.com/Winetricks/winetricks


You will need to update box86 and box64 to their last versions and use bash as written here:

https://box86.org/2022/09/running-bash-with-box86-box64/

TLDR: Download bash x86 and bash x64 from box86 and box64 github repo test folder and put them in a folder in your home directory. In terminal execute:

export BOX64_BASH=~/folder/nameofbash_x64
export BOX86_BASH=~/folder/nameofbash_x86

Execute winetricks with this command:

WINEPREFIX=~/.wine32(or whatever your 32bit prefix is called) WINEARCH=win32 box64 winetricks
